six/modules/c++/six.sicd/source/Utilities.cpp

```diff
@@ -784,6 +877,194 @@ std::auto_ptr<ComplexData> Utilities::createFakeComplexData()
+    six::Poly2D outputXYToSlantX;               
+    six::Poly2D outputXYToSlantY;
+    six::Poly2D slantXYToOutputX;               
+    six::Poly2D slantXYToOutputY;
+    fitXYProjectionPolys(outputMesh,
+                         slantMesh,
+                         orderX,
+                         orderY,
+                         outputXYToSlantX,                  
+                         outputXYToSlantY,                  
+                         slantXYToOutputX,               
+                         slantXYToOutputY);
+
+    transformXYProjectionPolys(outputXYToSlantX,
+                               outputXYToSlantY,
+                               slantSampleSpacing,
+                               outputSampleSpacing,
+                               slantCenter,
+                               outputCenter,
+                               outputRowColToSlantRow,
+                               outputRowColToSlantCol);
+}
+
+six::Poly2D Utilities::transformXYPolyToRowColPoly(
+    const six::Poly2D& polyXY,
+    const types::RowCol<double>& outSampleSpacing,
+    const types::RowCol<double>& outCenter,
+    double polyScaleFactor,
+    double polyShift)
+{
+    const six::Poly2D xToRowTransform =
+        getXYtoRowColTransform(outCenter.row, outSampleSpacing.row, true);
+
+    const six::Poly2D yToColTransform =
+        getXYtoRowColTransform(outCenter.col, outSampleSpacing.col, false);    
+
+    // Transform the output to input polynomials to 
+    // take row, column inputs
+    six::Poly2D polyRowCol = polyXY.transformInput(
+        xToRowTransform, yToColTransform);
+
+    polyRowCol *= polyScaleFactor;
+    polyRowCol[0][0] += polyShift;
+
+    return polyRowCol;
+}
+
+void Utilities::transformXYProjectionPolys(
+    const six::Poly2D& outputXYToInputX,
+    const six::Poly2D& outputXYToInputY,
+    const types::RowCol<double>& inSampleSpacing,
+    const types::RowCol<double>& outSampleSpacing,
+    const types::RowCol<double>& inCenter,
+    const types::RowCol<double>& outCenter,
+    six::Poly2D& outputRowColToInputRow,
+    six::Poly2D& outputRowColToInputCol)
+{
+    outputRowColToInputRow = transformXYPolyToRowColPoly(
+        outputXYToInputX,
+        outSampleSpacing,
+        outCenter,
+        -1.0 / inSampleSpacing.row,
+        inCenter.row);
+    outputRowColToInputCol = transformXYPolyToRowColPoly(
+        outputXYToInputY,
+        outSampleSpacing,
+        outCenter,
+        -1.0 / inSampleSpacing.col,
+        inCenter.col);
+}
+
+void Utilities::fitXYProjectionPolys(
+    const six::sicd::PlanarCoordinateMesh& outputMesh,
+    const six::sicd::PlanarCoordinateMesh& slantMesh,
+    size_t orderX,
+    size_t orderY,
+    six::Poly2D& outputXYToSlantX,                  
+    six::Poly2D& outputXYToSlantY,                  
+    six::Poly2D& slantXYToOutputX,                  
+    six::Poly2D& slantXYToOutputY)              
+{
+    // Initial values.
+    outputXYToSlantX = six::Poly2D(0,0);
```

Comment:
These don't need to get initialized - they're always assigned below (unless we throw in which case that's fine that they're not initialized)